Sources/GRPC/ConnectionManagerChannelProvider.swift

```diff
@@ -76,32 +74,36 @@ internal struct DefaultChannelProvider: ConnectionManagerChannelProvider {
     // Making a `NIOSSLContext` is expensive and we should only do it (at most) once per TLS
     // configuration. We do it now and surface any error during channel creation (we're limited by
     // our API in when we can throw any error).
-    let sslContext: Result<NIOSSLContext, Error>? = configuration.tls.map { tls in
-      return Result {
-        try NIOSSLContext(configuration: tls.configuration)
-      }
+    //
+    // 'nil' means we're not using TLS, or we're using the Network.framework TLS backend. We'll
+    // check and apply the Network.framework TLS options when we create a bootstrap.
+    let sslContext: Result<NIOSSLContext, Error>?
```

Comment:
Hmm, `Optional<Result>` is sufficiently opaque that I'm never going to be quite sure what each thing means. I wonder if this needs a temporary type.

Reply:
Yeah, it is a little unclear. I extracted this into a `TLSMode`.